File: springboot/src/main/java/com/edufelip/meer/domain/auth/ForgotPasswordUseCase.java
package com.edufelip.meer.domain.auth;

import com.edufelip.meer.domain.repo.AuthUserRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.UUID;

public class ForgotPasswordUseCase {
    private static final Logger log = LoggerFactory.getLogger(ForgotPasswordUseCase.class);
    private final AuthUserRepository authUserRepository;

    public ForgotPasswordUseCase(AuthUserRepository authUserRepository) {
        this.authUserRepository = authUserRepository;
    }

    public void execute(String email) {
        var user = authUserRepository.findByEmail(email);
        if (user == null) {
            return; // avoid enumeration
        }
        String token = UUID.randomUUID().toString();
        log.info("Generated password reset token for user {}: {}", user.getEmail(), token);
    }
}

File: springboot/src/main/kotlin/com/edufelip/meer/domain/auth/ForgotPasswordUseCase.kt
package com.edufelip.meer.domain.auth

import com.edufelip.meer.domain.AuthUserRepository
import org.slf4j.LoggerFactory
import java.util.UUID

class ForgotPasswordUseCase(
    private val authUserRepository: AuthUserRepository
) {
    private val log = LoggerFactory.getLogger(ForgotPasswordUseCase::class.java)

    fun execute(email: String) {
        val user = authUserRepository.findByEmail(email)
        if (user == null) {
            // Avoid enumeration: respond success even if the email is not registered.
            return
        }

        val resetToken = UUID.randomUUID().toString()
        // TODO: Persist reset token and send email. For now, log for visibility during development.
        log.info("Generated password reset token for user {}: {}", user.email, resetToken)
    }
}
